examples/golang/topic-creation/topic-creation.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX - License - Identifier: Apache - 2.0
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+)
+
+// SNSCreateTopicAPI defines the interface for the CreateTopic function.
+// We use this interface to test the function using a mocked service.
+type SNSCreateTopicAPI interface {
```

Comment:
why do we need this interface?

Reply:
We do not need it. I just took it from the example code. Sorry about it.